src/libraries/System.Text.Json/tests/Serialization/EnumConverterTests.cs

```diff
@@ -294,5 +312,212 @@ public enum MyEnum
             U = 1 << 20,
             V = 1 << 21,
         }
+
+        [Fact, OuterLoop]
+        public static void VeryLargeAmountOfEnumDictionaryKeysToSerialize()
```

Comment:
How long does this test end up taking (and roughly how much memory does it use up)?

Reply:
@Jacksondr5 Correct me if I'm wrong but I think it instantiates 2,097,152 (`1 << 21`) dictionaries. 

Is there something that we could do to reduce this number while keeping the coverage? Perhaps you could do `Serialize<MyEnum>` until you have the cache at the desired number and then call `Serialize<Dictionary<MyEnum, int>` on the desired edge cases.

Reply:
I'll try my hand at benchmark.net and post some comparisons between `VeryLargeAmountOfEnumsToSerialize` and `VeryLargeAmountOfEnumDictionaryKeysToSerialize` so we can understand how long this takes and how serializing the `Dictionary<MyEnum,int>` compares to serializing `MyEnum`s.

Reply:
A rough estimate is sufficient (which probably doesn't need BDN, but feel free to go down that route). 

If you run the tests locally from master, and then compare the test time of your branch (remove the outerloop from the attribute to try it), that should give you a general idea. I am looking for an order of magnitude. If the tests went from taking ~1 minute, to 2+, or if we are using up 100s of MB/GB of memory (on Windows, Task manager could show it), then the test is too resource intensive, and we would definitely keep it as outerloop, and try to make it simpler, if possible.

If not, then outerloop is good enough. I asked to understand whether we need this test to be outerloop or if it is fast enough to become part of innerloop.

Reply:
Base time running `dotnet build /t:test`: 
```
=== TEST EXECUTION SUMMARY ===
    System.Text.Json.Tests  Total: 8715, Errors: 0, Failed: 0, Skipped: 0, Time: 33.520s
```

Time running after adding this new test:
```
=== TEST EXECUTION SUMMARY ===
    System.Text.Json.Tests  Total: 8716, Errors: 0, Failed: 0, Skipped: 0, Time: 40.564s
```

@ahsonkhan I would consider keeping this test as `OuterLoop` considering that the test above, `VeryLargeAmountOfEnumsToSerialize`, which uses a similar code minus the insane amount of dictionary instances, is 
 already signaled to run on `OuterLoop`.

Reply:
> Time: 33.520s
> Time: 40.564s

> I would consider keeping this test as `OuterLoop` considering that the test above, `VeryLargeAmountOfEnumsToSerialize`, which uses a similar code minus the insane amount of dictionary instances, is already signaled to run on `OuterLoop`.

Agreed. That sounds good.